android/src/reactNativeVersionPatch/nativeHierarchyManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java

```diff
@@ -254,14 +254,32 @@ public synchronized void updateLayout(
       if (container != null
           && viewManagerName.equals("RNSScreen")
           && mReaLayoutAnimator != null) {
-        ((ReaLayoutAnimator) mReaLayoutAnimator).getAnimationsManager().viewsDidLayout();
+        boolean hasHeader = topScreenHasHeader((ViewGroup) container);
+        if (!hasHeader || !container.isLayoutRequested()) {
+          mReaLayoutAnimator.getAnimationsManager().viewsDidLayout();
+        }
       }
     } catch (IllegalViewOperationException e) {
       // (IllegalViewOperationException) == (vm == null)
       e.printStackTrace();
     }
   }
 
+  private boolean topScreenHasHeader(ViewGroup screenStack) {
```

Comment:
The name of this method sounds more like a property, an ideal method should start with a verb. However, I cannot find a better name so I'm okay with the current name.